README.md

```diff
@@ -4,111 +4,136 @@ An API for validating survey schemas.
+curl -sSL https://install.python-poetry.org | python3 - --version 2.1.2
+poetry install
+```
 
+## Running locally
 
 To run the app:
-
-```
+``` shell
 make run
 ```
+
+Validator defaults to running on `http://localhost:5002/validate`.
+
+You can override this by setting the `AJV_VALIDATOR_SCHEME` , `AJV_VALIDATOR_HOST`, and `AJV_VALIDATOR_PORT` environment variables.
+The defaults for these are:
+- `AJV_VALIDATOR_SCHEME` = "http"
+- `AJV_VALIDATOR_HOST` = "localhost"
+- `AJV_VALIDATOR_PORT` = "5002"
+
+Alternatively, you can override the entire URL by setting the `AJV_VALIDATOR_URL` environment variable directly.
+
 If you want to run the app locally using multiple server workers set reload to "False" in the Uvicorn settings in api.py:
-```python
+``` python
 uvicorn.run("api:app", workers=20, port=5001, reload=False)
 ```
-The validator can be called directly in the browser using the "/validate" endpoint and the "url" parameter for the address where the schema is located (eg. GitHub Gist raw json). Only URLs from the ONSDigital organization are accepted.
-```
-http://localhost:5001/validate?url=...
-```
 
+### Running against a URL
 
-## Testing
+Once validator is running, it can be called directly in the browser using the "/validate" endpoint and the "url" parameter for the address where the schema is located (eg. GitHub Gist raw json).
 
-By default, all schemas in the `tests/schemas/valid` directory will be evaluated as part of the unit tests.
-Any errors in these schemas will cause a failure.
+Only the following ur URLs and domains are accepted:
+- "https://gist.githubusercontent.com/"
+- "https://raw.githubusercontent.com/"
+- "onsdigital.uk"
+- "localhost"
 
-To run the app's unit tests:
+Also when using a URL from GitHub you can only validate schemas from the ONSdigital organisation.
 
+Here is an example of a command you can use to validate a schema via a URL:
 ```
-make test-unit
+http://localhost:5001/validate?url=https://raw.githubusercontent.com/ONSdigital/eq-questionnaire-runner/refs/heads/main/schemas/test/en/test_address.json
 ```
 
-To run the app's unit tests and ajv validator tests:
+### Running against eQ Runner
 
-```
-make test
-```
+Also once you have validator running it can be used to run against eQ runner (`https://github.com/ONSdigital/eq-questionnaire-runner`). If you have runner spun up you can run the `make validate-test-schemas` script from within the root of runner which will run validator on the test runner schemas.
 
-To test the apps functionality:
-```
-make run
-```
+### Running the Ajv (server) version of validator
 
-Then, in another terminal window/tab, navigate to a checked out copy of eq-survey-runner:
-```
-make test
-```
-
-# Installing node dependencies
+Included is a node based version of the json schema validation which may be used during development to assist with debugging errors. This returns more errors than we'd currently like due to the way polymorphism works for each of our blocks.
```

Comment:
I'm not sure this is really helpful, might be slightly confusing to someone new to the repo. If we are going to go down the route of talking about the errors being returned by our validators then may be worth adding some example outputs. Applies to line 96 below too.